### DISCUSSION TITLE
Filters Position, Speed, Sunrise

### DISCUSSION THREAD

**Discussion by Kaetzy** [maca88/SmartBikeLights#58] [Q&A]
Super cool app!!!

I want my Varia to go into day blink mode at the position. Unfortunately nothing happens. Is a cell phone connection necessary? (I tested it with a cell phone)

###0,73535488::16711680#2,2!Arbeit:1:7:0:0F1,51.05284,13.753917,51.053326,13.755827,51.054418,13.754615,51.053865,13.753112!:1:5:0:0D=1##2,2:Varia 515:65280:16777215!1,Solid:4!1,Night Flasch:6#0::#0:0#123!:3!4,6,7#255#0#B3570##3#0#0

**Comment by maca88** [comment]
Your configuration seems fine and it should work.

For the "Position" filter to work, you need to make sure that:
1. The light is in Smart mode. You should see the letter "S" inside the light icon, or "S" inside the top left button if you use 1 field layout
2. Edge should have a GPS signal

> Is a cell phone connection necessary?

No, it is not necessary. The data field works without a cell phone.

**Comment by Kaetzy** [comment]
Thanks for helping!

I also set it so that I only have mode “M” for light or looking, nothing more.
The goal is: normal light and looking in the specific position

Do you only use the filter in “S” Smart mode?

**Reply by maca88** [reply]
> I also set it so that I only have mode “M” for light

I missed that. You need to add also the Smart mode as filters do not work with "M" (Manual) mode. 

> normal light and looking in the specific position

You can achieve that only with the Smart mode.

**Comment by Kaetzy** [comment]
Thanks
I changed the mode "M" Manual and now mode "S" Smart,
everything works.

Now I have added speed, so now the position and the speed have to be right. I hope it works like that

###0,73535488::16711680#3,2!Work:2:6:0:0F2,51.086919,13.739433,51.086879,13.737307,51.093643,13.736499,51.093685,13.73776,51.10 4748,13.735667,51.104707,13.733844,51.108883, 13.732545,51.109503,13.734348C[5.55556!:1:4:0:0D=1##2,2:Varia 515:65280:16777215!1,Solid:4!1,Night Flasch:6#0::#0 :0#123!:1!#255#0#B3570##3#0#0

**Reply by maca88** [reply]
The configuration has whitespaces that are not allowed, use this one instead:

###0,73535488::16711680#3,2!Work:2:6:0:0F2,51.086919,13.739433,51.086879,13.737307,51.093643,13.736499,51.093685,13.73776,51.104748,13.735667,51.104707,13.733844,51.108883,13.732545,51.109503,13.734348C[5.55556!:1:4:0:0D=1##2,2:Varia515:65280:16777215!1,Solid:4!1,NightFlasch:6#0::#0:0#123!:1!#255#0#B3570##3#0#0

it is the same, but without whitespaces. The added speed filter is fine.

**Comment by Kaetzy** [comment]
Hey,
Thanks for the info about the blank lines, everything works now.

When I turn on the navigation system and the radar and then go to the bike profile, it always says "M" mode. I don't have "M" mode in it. Why is that ?

Programming the app and everything like that is really cool, but you definitely have to have a lot of background knowledge because just programming the app isn't enough.
​


###0,73535488::16711680#3,2!Work:2:6:0:0F3,51.086919,13.739433,51.086879,13.737307,51.093643,13.736499,51.093685,13.73776,51.103892,13.735646,51.103832,13.734112,51.108743,13.732578,51.108972,13.734659,51.113313,13.728093,51.113151,13.727701,51.112026,13.728796,51.112232,13.729096C[5.55556!:1:4:0:0D=1##2,2:Varia515:65280:16777215!1,Solid:4!1,NightFlasch:6#0::#0:0#123!:1!#255#0#B3570##3#0#0

**Comment by maca88** [comment]
> When I turn on the navigation system and the radar and then go to the bike profile, it always says "M" mode. I don't have "M" mode in it. Why is that ?

With `Lights icon tap behavior` you only specify which modes can be activated by tapping on the icon. It is still possible to move into Manual "M" mode in case tapping on one of the buttons on the light panel (Solid/Night Flash). The light will move to "M" mode also when switching the light off/on by using the physical button on the light. You can enable `Force Smart Mode` if you want to prevent moving into "M" mode when pressing the light physical button, but do note that you won't be able to turn off the light with the current configuration until Edge is on. I would propose to add a global filter, so that the light is turned off after the activity is finished and enable `Force Smart Mode` to prevent accidental moving to "M" mode. Here is the proposed configuration:

1,1!:1H]0###0,73535488::16711680#3,2!Work:2:6:0:0F3,51.086919,13.739433,51.086879,13.737307,51.093643,13.736499,51.093685,13.73776,51.103892,13.735646,51.103832,13.734112,51.108743,13.732578,51.108972,13.734659,51.113313,13.728093,51.113151,13.727701,51.112026,13.728796,51.112232,13.729096C[5.55556!:1:4:0:0D=1##2,2:Varia515:65280:16777215!1,Solid:4!1,NightFlasch:6#0::#0:1#123!:1!#255#0#B3570##3#0#0

**Comment by Kaetzy** [comment]
I can get back to "M" by clicking on the two buttons on the data page.
If I switch the navigation system off in "M" mode and switch it on again, I'm in "M" mode and not in "S" mode.

It's no problem that the light doesn't go out anymore,
because when I arrive, I always turn off the radar using the button

**Comment by Kaetzy** [comment]
Hey,

So a question to set up.
The lamp is set to a “default mode”. Then I can change the light through the filters if the condition is met, so far everything is clear.

I want to do a day and a night mode:
When the light is always on and flashing in certain places at night (I have it now)
During the day the light should be off and flashing in certain places.

But since “Default mode” cannot be changed, how do I do something like that?

###0,73535488::16711680#7,4!Nacht:3:6:0:0Es0,r0F3,51.086919,13.739433,51.086879,13.737307,51.093643,13.736499,51.093685,13.73776,51.103892,13.735646,51.103832,13.734112,51.108743,13.732578,51.108972,13.734659,51.113313,13.728093,51.113151,13.727701,51.112026,13.728796,51.112232,13.729096C[5.55556!Tag:1:0:0:0Er0,s0!TagB:2:7:0:0F1,51.093846,13.735721,51.09363,13.738811,51.086164,13.738339,51.087161,13.736622Er0,s0!:1:4:0:0D=1##2,2:Varia515:65280:16777215!1,Solid:4!1,NightFlasch:6#0::#0:1#123!:1!#255#0#B3570##3#0#0

**Reply by maca88** [reply]
> If I switch the navigation system off in "M" mode and switch it on again, I'm in "M" mode and not in "S" mode.

That is by design. The "Force Smart Mode" only works when in the "Smart" mode. If you manually change from "S" to "M" then that mode will be the same even after a device turn off & on.

> because when I arrive, I always turn off the radar using the button

That's why I've proposed to add the global filter, so that when you stop the activity the lights will automatically turn off, without having to manually turn them off.

> But since “Default mode” cannot be changed, how do I do something like that?

You just need to move filter group "TabB" before group "Tag" and it will work. The logic go through the filter groups from top to bottom. The first group that satisfies all conditions will be activated. Updated configuration:

###0,73535488::16711680#7,4!Nacht:3:6:0:0Es0,r0F3,51.086919,13.739433,51.086879,13.737307,51.093643,13.736499,51.093685,13.73776,51.103892,13.735646,51.103832,13.734112,51.108743,13.732578,51.108972,13.734659,51.113313,13.728093,51.113151,13.727701,51.112026,13.728796,51.112232,13.729096C[5.55556!TagB:2:7:0:0F1,51.093846,13.735721,51.09363,13.738811,51.086164,13.738339,51.087161,13.736622Er0,s0!Tag:1:0:0:0Er0,s0!:1:4:0:0D=1##2,2:Varia515:65280:16777215!1,Solid:4!1,NightFlasch:6#0::#0:1#123!:1!#255#0#B3570##3#0#0

**Comment by Kaetzy** [comment]
This is going great! Thanks
When the sun rises it's not quite light yet, so I would now program the "Peloton" 30 minutes after sunrise and 30 minutes before sunset.
But since I already have a filter with sunrise to sunset with "lights off", that doesn't work, does it?
Because if I set the sunrise to sunset “Peloton” filter correctly, it will only be done in the order and will not be added to the others

**Reply by maca88** [reply]
Yes, it can work if you put a new sunrise and sunset filter group before the "lights off". Here is how you do it:

###0,73535488::16711680#9,6!Nacht:3:6:0:0Es0,r0F3,51.086919,13.739433,51.086879,13.737307,51.093643,13.736499,51.093685,13.73776,51.103892,13.735646,51.103832,13.734112,51.108743,13.732578,51.108972,13.734659,51.113313,13.728093,51.113151,13.727701,51.112026,13.728796,51.112232,13.729096C[5.55556!Dawn:1:5:0:0Er0,r1800!Dusk:1:5:0:0Es-1800,s0!TagB:2:7:0:0F1,51.093846,13.735721,51.09363,13.738811,51.086164,13.738339,51.087161,13.736622Er0,s0!Tag:1:0:0:0Er0,s0!:1:4:0:0D=1##2,2:Varia515:65280:16777215!1,Solid:4!1,NightFlasch:6#0::#0:1#123!:1!#255#0#B3570##3#0#0

I've added Dusk and Dawn filter groups that will be active 30 minutes after sunrise and 30 minutes before sunset

**Comment by Kaetzy** [comment]
I have now changed it so that it works in economy mode (Peloton) and daytime viewing

###0,73535488::16711680#10,6!Nacht:3:6:0:0Es0,r0F3,51.086919,13.739433,51.086879,13.737307,51.093643,13.736499,51.093685,13.73776,51.103892,13.735646,51.103832,13.734112,51.108743,13.732578,51.108972,13.734659,51.113313,13.728093,51.113151,13.727701,51.112026,13.728796,51.112232,13.729096C[5.55556!TagBli:3:7:0:0Er0,s0F3,51.093655,13.736515,51.093689,13.73777,51.086859,13.739133,51.086879,13.737298,51.113158,13.727691,51.113296,13.728098,51.112225,13.729112,51.11205,13.72879,51.108735,13.732588,51.108978,13.734702,51.103892,13.735571,51.103906,13.734396C[5.55556!Aufgan:1:5:0:0Er0,r1200!Unterg:1:5:0:0Es-1200,s0!TagAus:1:0:0:0Er300,s-300!:1:4:0:0D=1##2,2:Varia515:65280:16777215!1,Solid:4!1,NightFlasch:6#0::#0:0#123!:1!#255#0#B3570##3#0#0
